telemetry_core/src/main/java/com/newrelic/telemetry/SpanBatchSenderFactory.java

```diff
@@ -4,18 +4,17 @@
  */
 package com.newrelic.telemetry;
 
+import com.newrelic.telemetry.http.HttpPoster;
 import com.newrelic.telemetry.spans.SpanBatchSender;
 import com.newrelic.telemetry.spans.SpanBatchSenderBuilder;
 import java.time.Duration;
 
 /**
- * A builder class for creating a SpanBatchSender that uses okhttp as the underlying http client
- * implementation.
+ * A factory interface for creating a SpanBatchSender.
  *
- * <p>Note: This class will be deprecated in the near future and be replaced with something named
- * more succinctly.
+ * <p>Concrete implementations use different HTTP providers.
  */
-public class SimpleSpanBatchSender {
```

Comment:
same comment. We can't delete this piece of the public API without at least a lot of pre-warning via a `@Deprecated` annotation and comments. I'd like to favor only accreting onto our public API, as long as the thing we're deprecating isn't actually broken/non-functional, even if it's not the API that we thing is the greatest in the long run.

Reply:
They're not deleted - I've converted them into interfaces, so I can sidestep the static state problem.

Reply:
renaming the class and replacing it with an interface is effectively deleting the class, from the user's perspective, no?